Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift

```diff
@@ -0,0 +1,485 @@
+//
+// Copyright (C) 2024 Apple Inc. All rights reserved.
+//
+
+import Foundation
+
+#if canImport(WritingTools)
+
+#if canImport(AppKit)
+import AppKit
+// WritingToolsUI is not present in the base system, but WebKit is, so it must be weak-linked.
+// WritingToolsUI need not be soft-linked from WebKitSwift because although WTUI links WebKit, WebKit does not directly link WebKitSwift.
+@_weakLinked internal import WritingToolsUI_Private._WTTextEffectView
+@_weakLinked internal import WritingToolsUI_Private._WTSweepTextEffect
+@_weakLinked internal import WritingToolsUI_Private._WTReplaceTextEffect
+#else
+internal import UIKit_Private
+@_spi(TextEffects) import UIKit
+#endif
+
+import WebKitSwift
+
+// MARK: Platform abstraction type aliases
+
+#if canImport(AppKit)
+typealias PlatformView = NSView
```

Comment:
> I'm guessing the C++ type aliases defined in PlatformView.h can't be reused?
nope

> Have we thought about setting up some platform abstractions at large for WebKitSwift? Would a WebKitSwift/platform/cocoa/PlatformView.swift file make sense with this content make sense? This suggestion may be completely off-key so feel free to ignore.

This would probably be worth doing if WebKitSwift is going to be long lived, but since we’re trying to migrate to being able to write Swift in WebKit, doing this now would make such a migration a bit more painful

Reply:
Oh that's good to know, maybe we can have `PlatformView.swift` live next to `PlatformView.h` when we can _just_ write Swift in WebKit.